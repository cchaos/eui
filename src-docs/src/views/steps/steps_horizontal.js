import React from 'react';

import {
  EuiStepsHorizontal,
} from '../../../../src/components';

const horizontalSteps = [
  {
    title: 'Completed Step 1',
    isComplete: true,
  },
  {
    title: 'Selected Step 2',
    isSelected: true,
  },
  {
    title: 'Incomplete Step 3',
  },
  {
    title: 'Disabled Step 4',
    disabled: true,
  },
];

export default () => (
  <div>
    <EuiStepsHorizontal
      steps={horizontalSteps}
    />
  </div>
);
