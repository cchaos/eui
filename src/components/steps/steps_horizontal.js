import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export const EuiStepsHorizontal = ({
  children,
  className,
  ...rest
}) => {
  const classes = classNames(
    'euiStepsHorizontal',
    className
  );

  return (
    <div
      role="tablist"
      className={classes}
      {...rest}
    >
      {children}
    </div>
  );
};

EuiStepsHorizontal.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};
