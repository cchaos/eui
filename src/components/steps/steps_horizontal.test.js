import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test';

import { EuiStepsHorizontal } from './steps_horizontal';

describe('EuiStepsHorizontal', () => {
  test('is rendered', () => {
    const component = render(
      <EuiStepsHorizontal {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
