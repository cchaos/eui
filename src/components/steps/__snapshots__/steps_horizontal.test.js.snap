// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiStepsHorizontal is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiStepsHorizontal testClass1 testClass2"
  data-test-subj="test subject string"
  role="tablist"
/>
`;
